workspaces/rbac/plugins/rbac-backend/src/service/policies-rest-api.ts

```diff
@@ -893,11 +1041,34 @@ export class PoliciesServer {
           throw new NotAllowedError(); // 403
         }
 
+        if (decision.result === AuthorizeResult.CONDITIONAL) {
+          conditionsFilter = transformConditions(decision.conditions);
+        }
+
         const id: number = parseInt(request.params.id, 10);
         if (isNaN(id)) {
           throw new InputError('Id is not a valid number.');
         }
 
+        const condition = await this.conditionalStorage.getCondition(id);
+
+        if (!condition) {
+          throw new NotFoundError(`Condition with id ${id} was not found`);
+        }
+
+        const oldCondition: RoleConditionalPolicyDecision<PermissionAction> = {
```

Comment:
Minor: I think you can remove this oldCondition object. Then you can use `condition.roleEntityRef` to retrieve role metadata. Additinally you can rename condition to oldCondition. What do you think @PatAKnight ?

Reply:
Fixed